server/sonar-server/src/main/java/org/sonar/server/rule/ws/RuleMapper.java

```diff
@@ -78,7 +78,14 @@ public RuleMapper(final Languages languages, final MacroInterpreter macroInterpr
   public Rules.Rule toWsRule(RuleDto ruleDto, SearchResult result, Set<String> fieldsToReturn) {
     Rules.Rule.Builder ruleResponse = Rules.Rule.newBuilder();
 
+    // Mandatory fields
     ruleResponse.setKey(ruleDto.getKey().toString());
+    Common.RuleType type = Common.RuleType.valueOf(ruleDto.getType());
+    if (type != null) {
```

Comment:
ok, @teryk reminded me that the goal is to stop supporting optional response fields.